cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -170,7 +170,7 @@ func NewSupervisor(logger *zap.Logger, configFile string) (*Supervisor, error) {
 	}
 
 	if connErr := s.waitForOpAMPConnection(); connErr != nil {
-		return nil, fmt.Errorf("failed to connect to the OpAMP server: %w", connErr)
+		logger.Debug("failed to connect to the OpAMP server.", zap.Error(connErr))
```

Comment:
it doesn't seem like the supervisor retries to connect here. Wouldn't that hinder the opampsupervisor useless for its lifetime?

Reply:
the connection attempts are started in `startOpAMP`, a few lines before this one. 

Note this test, where the connection succeeds even after `waitForOpAMPConnection` 'gives up waiting':
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33275/files#diff-239707dd27c36360814b762f18855fdb907bca784df484a72e90d6106daf3de6R734-R738

The purpose for this change is:
1. If the supervisor cannot connect to a server, it should not surrender and kill itself+agent. Instead, it should run with the latest config. **However**, it will continue to try to connect (as tested by the test above)
2. If the supervisor is able to connect to a server, then things will behave as expected

Reply:
one more note - the function `waitForOpAMPConnection` only synchronizes the current goroutine with the goroutine in charge of connecting with the OpAMP server. It's not the function in charge of connecting.

Reply:
ah i see now, thanks for explaining. [the WS client indeed continually retries](https://github.com/open-telemetry/opamp-go/blob/main/client/wsclient.go#L186).